packages/editor/src/bindings/post-meta.js

```diff
@@ -9,26 +9,65 @@ import { store as coreDataStore } from '@wordpress/core-data';
 import { store as editorStore } from '../store';
 import { unlock } from '../lock-unlock';
 
-function getMetadata( registry, context, registeredFields ) {
-	let metaFields = {};
-	const type = registry.select( editorStore ).getCurrentPostType();
+/**
+ * Get a list of post meta fields with their values and labels
+ * to be consumed in the needed callbacks.
+ * If the value is not available based on context, like in templates,
+ * it falls back to the default value, label, or key.
+ *
+ * @param {Object} registry The registry context exposed through `useRegistry`.
+ * @param {Object} context  The context provided.
+ * @return {Object} List of post meta fields with their value and label.
+ *
+ * @example
+ * ```js
+ * {
+ *     field_1_key: {
+ *         label: 'Field 1 Label',
+ * 	       value: 'Field 1 Value',
+ *     },
+ *     field_2_key: {
+ *         label: 'Field 2 Label',
+ * 	       value: 'Field 2 Value',
+ *     },
```

Comment:
Nit: I'm not sure whether spaces or tabs should be used, but it should be at least consistent:
```suggestion
 *     field_1_key: {
 *         label: 'Field 1 Label',
 *         value: 'Field 1 Value',
 *     },
 *     field_2_key: {
 *         label: 'Field 2 Label',
 *         value: 'Field 2 Value',
 *     },
```